crates/storage/provider/src/bundle_state/hashed_state_changes.rs

```diff
@@ -0,0 +1,69 @@
+use reth_db::{
+    cursor::{DbCursorRO, DbCursorRW, DbDupCursorRO, DbDupCursorRW},
+    tables,
+    transaction::{DbTx, DbTxMut},
+    DatabaseError,
+};
+use reth_primitives::{Account, StorageEntry, B256, U256};
+use reth_trie::hashed_cursor::HashedPostState;
+use std::collections::BTreeMap;
+
+/// Changes to the hashed state.
+#[derive(Debug, Default)]
+pub struct HashedStateChanges(pub HashedPostState);
+
+impl HashedStateChanges {
+    /// Write the bundle state to the database.
+    pub fn write_to_db<TX: DbTxMut + DbTx>(self, tx: &TX) -> Result<(), DatabaseError> {
```

Comment:
it would be nice to add the timings breakdown here too, same as in `insert_hashes`

Reply:
also a bit concerning that this is almost the same logic as in `insert_(account|storage)_for_hashing`

Reply:
will address in a follow up